services/transactionService.ts

```diff
@@ -595,3 +612,205 @@ export async function getOldestPositiveTxForUser (userId: string): Promise<Trans
+      LEFT JOIN \`PricesOnTransactions\` pt ON t.\`id\` = pt.\`transactionId\`
+      LEFT JOIN \`Price\` pb ON pt.\`priceId\` = pb.\`id\`
+      WHERE t.\`amount\` > 0
+      AND EXISTS (
+        SELECT 1
+        FROM \`AddressesOnUserProfiles\` au
+        WHERE au.\`addressId\` = a.\`id\`
+          AND au.\`userId\` = ${userId}
+      )
+      AND p.\`providerUserId\` = ${userId}
+      GROUP BY t.id, p.id
+      ORDER BY p.\`name\` DESC
+      LIMIT ${pageSize}
+      OFFSET ${offset};
+    `
+  }
+
+  const payments: Payment[] = []
+
+  transactions.forEach((tx: any) => {
+    const ret: QuoteValues = {
+      usd: new Prisma.Decimal(0),
+      cad: new Prisma.Decimal(0)
+    }
+    for (const p of JSON.parse(tx.prices)) {
+      if (p.quoteId === USD_QUOTE_ID) {
+        ret.usd = ret.usd.plus(p.priceValue * tx.amount)
+      }
+      if (p.quoteId === CAD_QUOTE_ID) {
+        ret.cad = ret.cad.plus(p.priceValue * tx.amount)
+      }
+    }
+    const buttonDisplayDataList: ButtonDisplayData[] = []
+    buttonDisplayDataList.push({
+      name: tx.paybuttonName,
+      id: tx.paybuttonId,
+      providerUserId: tx.paybuttonProviderUserId
+    })
+    if (tx.amount > 0) {
+      payments.push({
+        timestamp: tx.timestamp,
+        values: {
+          values: ret,
+          amount: tx.amount
+        },
+        networkId: tx.networkId,
+        hash: tx.hash,
+        buttonDisplayDataList
+      })
+    }
+  })
+
+  return payments
+}
+
+export async function fetchAllPaymentsByUserIdWithPagination (
+  userId: string,
+  page: number,
+  pageSize: number,
+  orderBy?: string,
+  orderDesc = true
+): Promise<Payment[]> {
+  const orderDescString: Prisma.SortOrder = orderDesc ? 'desc' : 'asc'
+
+  let orderByQuery
+  if (orderBy !== undefined && orderBy !== '') {
+    if (orderBy === 'values') {
+      orderByQuery = {
+        amount: orderDescString
+      }
+    } else if (orderBy === 'networkId') {
+      orderByQuery = {
+        address: {
+          networkId: orderDescString
+        }
+      }
+    } else {
+      orderByQuery = {
+        [orderBy]: orderDescString
+      }
+    }
+  } else {
+    // Default orderBy
+    orderByQuery = {
+      timestamp: orderDescString
+    }
+  }
+
+  const transactions = await prisma.transaction.findMany({
+    where: {
+      address: {
+        userProfiles: {
+          some: {
+            userId
+          }
+        }
+      },
+      amount: {
+        gt: 0
+      }
+    },
+    include: includePaybuttonsAndPrices,
+    orderBy: orderByQuery,
+    skip: page * Number(pageSize),
+    take: Number(pageSize)
+  })
+
+  const transformedData: Payment[] = []
+  for (let index = 0; index < transactions.length; index++) {
+    const tx = transactions[index]
+    if (Number(tx.amount) > 0) {
+      const payment = await generatePaymentFromTx(tx)
+      transformedData.push(payment)
+    }
+  }
+  return transformedData
+}
+
+export async function fetchAllPaymentsByUserId (
```

Comment:
I feel this function shouldn't exist. The `if` could just be inside the `fetchAllPaymentsByUserIdWithPagination`, which is already a better name for what this function, given what it does, and then no else would be required.